### PR TITLE
cmake: Make building of tests optional

### PR DESCRIPTION
**Pull request by GBuella**
Introduce the BUILD_LIBPMEMFILE_POSIX_TESTS and
BUILD_LIBPMEMFILE_TESTS options.

---
This change is [Reviewable](https://reviewable.io/reviews/pmem/pmemfile/339)

### CODE REVIEW

**Comment by krzycz** [comment]
Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/pmemfile/339)*

File: CMakeLists.txt [hunk 1]
```diff
@@ -46,6 +46,15 @@ include(FindThreads)
 option(BUILD_LIBPMEMFILE
 	"build libpmemfile (requires libsyscall_intercept)" ON)
 
+option(BUILD_LIBPMEMFILE_TESTS
+	"build libpmemfile tests (ignored if BUILD_LIBPMEMFILE is not set)" ON)
+
+if(BUILD_LIBPMEMFILE AND BUILD_LIBPMEMFILE_TESTS)
+	set(ACTUALLY_BUILD_LIBPMEMFILE_TESTS ON)
```
Comment (marcinslusarz) [inline CMakeLists.txt]: With
```
if (NOT BUILD_LIBPMEMFILE)
   set(BUILD_LIBPMEMFILE_TESTS OFF)
endif()
```
you won't need ACTUALLY_BUILD_LIBPMEMFILE_TESTS.

File: CMakeLists.txt [hunk 2]
```diff
@@ -454,10 +463,12 @@ if(NOT LIBUNWIND_FOUND)
 	message(WARNING "libunwind not found. Stack traces from tests will not be reliable")
 endif()
 
-if(TEST_DIR)
-	enable_testing()
-else()
-	message(WARNING "TEST_DIR is empty - 'make test' will not work")
+if(BUILD_LIBPMEMFILE_POSIX_TESTS OR ACTUALLY_BUILD_LIBPMEMFILE_TESTS)
```
Comment (marcinslusarz) [inline CMakeLists.txt]: Please make a variable for that (TESTS_ENABLED?).

File: CMakeLists.txt (hunk 1 above)
Comment (GBuella) [reply CMakeLists.txt]: You can't just override an option set by the user (I tried it, it just stays ON).
But I'll try to use CMakeDependentOption instead.

Comment (GBuella) [reply CMakeLists.txt]: Done

File: CMakeLists.txt (hunk 2 above)
Comment (GBuella) [reply CMakeLists.txt]: Done

File: tests/CMakeLists.txt [hunk 3]
```diff
@@ -29,7 +29,9 @@
 # (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 # OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
-option(TESTS_USE_FORCED_PMEM "let tests force enable or force disable use of optimized flush in libpmemobj (to speed them up)" OFF)
+cmake_dependent_option(TESTS_USE_FORCED_PMEM
+	"let tests force enable or force disable use of optimized flush in libpmemobj (to speed them up)" OFF
+	"BUILD_LIBPMEMFILE_POSIX_TESTS;BUILD_LIBPMEMFILE_TESTS" OFF)
```
Comment (marcinslusarz) [inline tests/CMakeLists.txt]: Are you sure this is correct? The way I read the CMakeDependentOption documentation this makes TESTS_USE_FORCED_PMEM depend on *both* BUILD_LIBPMEMFILE_POSIX_TESTS and BUILD_LIBPMEMFILE_TESTS.

Comment (GBuella) [reply tests/CMakeLists.txt]: I'm not sure if this is correct.

Comment (GBuella) [reply tests/CMakeLists.txt]: Well, I guess disjunction is not supported by `cmake_dependent_option`, so I just reverted this part of the patch.

**Comment by krzycz** [comment]
Reviewed 1 of 2 files at r2, 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/pmemfile/339)*

**Comment by marcinslusarz** [comment]
:lgtm:

---

Reviewed 1 of 2 files at r2, 1 of 1 files at r3.
Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/pmemfile/339#-:-KtkmrZ4Riw9owmNqTXt:bnfp4nl)*

**Comment by sarahjelinek** [comment]
:lgtm:

---

Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/pmemfile/339#-:-Ktlpr0VaSNnmvd8IoF5:bnfp4nl)*

**Pull request merged**